### PR TITLE
Feedback round 2

### PR DESCRIPTION
**Pull request by geoffhull03**
Still unsure if the Revoke functionality is actually revoking the token and there are some login/cookie issues on the Procore side I'm not sure that we can (or necessarily need to) solve for.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,7 +25,7 @@ Within this file, configure your application's Client ID, Client Secret, and Red
 5. Navigate to the directory where the repository was cloned to and [launch your Rails server](https://guides.rubyonrails.org/command_line.html#rails-server).
 6. The landing page will include a button that says, "Sign In to Procore". Click this button and enter your Procore email address/password.
 7. After authenticating with Procore, you will be redirected back to the sample application. This page will include a table containing the first and last five characters of both your access token and your refresh token. In addition, there will be timestamps corresponding to when the access token was generated and when it expires (2 hours after generation).
-8. To access the data returned by the [Show User Info](https://developers.procore.com/reference/me) endpoint, click on the "Show User Information" button on the home page.
+8. To access the data returned by the [Show User Info](https://developers.procore.com/reference/me) endpoint, click on the "Show User Information" button on the home page. Note, the "name" field will only populate for a user when a
```
Comment (colinfujiiprocore) [inline README.md]: I think this line is unfinished

Comment (geoffhull03) [inline README.md]: Reverted the proposed change to item 8

File: app/controllers/login_controller.rb [hunk 2]
```diff
@@ -23,8 +23,15 @@ def callback
     session[:oauth_response] = JSON.parse(response)
 
     # Redirect the user's browser to the intended home page
-    redirect_to users_home_path
-  end
+    redirect_to users_home_path, success: "You have successfully obtained an access token!"
+
+    rescue RestClient::ExceptionWithResponse
+      if session[:oauth_response]
+        redirect_to users_home_path, danger: "Something Went Wrong. Please Refresh your Access Token and Try Again"
```
Comment (nick-morrill) [inline app/controllers/login_controller.rb]: Does every word in the error message need to be capitalized?

There should be period after "try again", too.

File: app/controllers/login_controller.rb [hunk 3]
```diff
@@ -23,8 +23,15 @@ def callback
...
+      else
+        redirect_to login_index_path, danger: "Something Went Wrong. Please try again"
```
Comment (nick-morrill) [inline app/controllers/login_controller.rb]: There should be a period after "try again".

File: app/controllers/login_controller.rb [hunk 4]
```diff
@@ -43,6 +50,37 @@ def refresh
     session[:oauth_response] = JSON.parse(response)
 
     # Redirect the user's browser to the intended home page
-    redirect_to users_home_path
+    redirect_to users_home_path, success: "You have successfully refreshed your access token!"
+
+    rescue RestClient::ExceptionWithResponse
+      if session[:oauth_response]
+        redirect_to users_home_path, danger: "Something Went Wrong. Please Refresh your Access Token and Try Again"
```
Comment (nick-morrill) [inline app/controllers/login_controller.rb]: My comments regarding the above error messages also apply to this one and the next one (on line 59).

File: app/controllers/login_controller.rb [hunk 5]
```diff
@@ -43,6 +50,37 @@ def refresh
...
+    # Note that in the production environment, this request uses the BASE_URL (https://api.procore.com) to revoke the token.
+    response = RestClient.post(ENV['BASE_URL'] + '/oauth/revoke', request.to_json)
+    # Deletes current session to clear out session variables
+    reset_session
+    # Redirects user back to the login page
+    redirect_to login_index_path
+
+    rescue RestClient::ExceptionWithResponse
+      if session[:oauth_response]
+        redirect_to users_home_path, danger: "Something Went Wrong. Please Refresh your Access Token and Try Again"
```
Comment (nick-morrill) [inline app/controllers/login_controller.rb]: My comments regarding the above error messages also apply to this one and the next one (on line 83).

File: app/controllers/users_controller.rb [hunk 6]
```diff
@@ -8,6 +8,13 @@ def me
 
     # Store the parsed response in an instance variable
     @me = JSON.parse(get_me)
+
+    rescue RestClient::ExceptionWithResponse
+      if session[:oauth_response]
+        redirect_to users_home_path, danger: "Something Went Wrong. Please Check/Refresh your Access Token and Try Again"
```
Comment (nick-morrill) [inline app/controllers/users_controller.rb]: My comments regarding the messages in `login_controller.rb` also apply to this message and the others within this file (lines 16, 43, and 45).

File: app/views/users/home.html.erb [hunk 7]
```diff
@@ -1,3 +1,11 @@
+<head>
+  <% flash.each do |key, value| %>
+    <div class='alert alert-<%= key %>' >
+      <%= value %>
+      </div>
```
Comment (nick-morrill) [inline app/views/users/home.html.erb:5]: I think the spacing is slightly off on this `</div>`

File: config/application-copy.yml [hunk 8]
```diff
@@ -0,0 +1,9 @@
+CLIENT_ID: ''
+CLIENT_SECRET: ''
+REDIRECT_URI: ''
+OAUTH_URL: ''
+BASE_URL: ''
+
+#Copy these keys into your followed application.yml file. The application.yml file is created after the installation of the Figaro gem (see Readme for more instructions)
```
Comment (nick-morrill) [inline config/application-copy.yml]: There should be a space at the start of each of these comment lines, after the `#`

File: config/routes.rb [hunk 9]
```diff
@@ -10,6 +10,10 @@
 
   get 'login/callback'
 
+  get 'login/revoke'
+
+  post 'login/revoke'
```
Comment (nick-morrill) [inline config/routes.rb]: Do you need both the GET and the POST route?

**Review by nick-morrill — DISMISSED** [review]
Are all the `tmp/cache/assets/sprockets/*` files necessary or can they be added to the .gitignore file?

File: config/routes.rb (hunk 9 above)
Comment (colinfujiiprocore) [reply config/routes.rb]: technically I believe the GET route being present helps prevent an issue when the user hits the back arrow and the browser attempts to navigate back to that page. Hitting the back arrow is automatically a GET request so if the user does hit back/forward and hits that route the app will throw an unknown route 500.

Comment (colinfujiiprocore) [reply config/routes.rb]: to be honest, this should probably just be a GET request, since the request isn't sending any data anyways.

**Review by colinfujiiprocore — APPROVED** [review]
I feel like there's probably another way to do the controller error handling, but I can't think of it for the moment.

**Pull request merged**